dev-libs/mongo-c-driver/mongo-c-driver-1.30.6.ebuild

```diff
@@ -0,0 +1,201 @@
+	sed -i 's#<bson/bson-iso8601-private.h>#"bson/bson-iso8601-private.h"#' src/libbson/tests/test-json.c || die
+	sed -i 's#<bson/bson-json-private.h>#"bson/bson-json-private.h"#' src/libbson/tests/test-json.c || die
+	sed -i 's#<bson/bson-context-private.h>#"bson/bson-context-private.h"#' src/libbson/tests/test-oid.c || die
+	sed -i 's#<bson/bson-iso8601-private.h>#"bson/bson-iso8601-private.h"#' src/libbson/tests/test-oid.c || die
+
+	# bug 953521
+	sed -i 's/message (FATAL_ERROR "System libbson built without static library target")/message (STATUS "System libbson built without static library target")/' CMakeLists.txt || die
+}
+
+src_configure() {
+	use static-libs && lto-guarantee-fat
+	local mycmakeargs=(
+		-DCMAKE_SKIP_RPATH=ON # mongoc-stat insecure runpath
+		-DUSE_SYSTEM_LIBBSON=ON
+		-DENABLE_CLIENT_SIDE_ENCRYPTION=OFF
+		-DENABLE_EXAMPLES=OFF
+		-DENABLE_MAN_PAGES="$(usex man ON OFF)"
+		-DENABLE_MONGOC=ON
+		-DENABLE_MONGODB_AWS_AUTH=OFF
+		-DENABLE_SNAPPY=ON
+		-DENABLE_ZLIB=SYSTEM
+		-DENABLE_SASL="$(usex sasl CYRUS OFF)"
+		-DENABLE_SRV=ON
+		-DENABLE_SSL="$(usex ssl $(usex ssl OPENSSL) OFF)"
+		-DENABLE_STATIC="$(usex static-libs ON OFF)"
+		-DENABLE_TESTS="$(usex test ON OFF)"
+		-DENABLE_TRACING="$(usex debug ON OFF)"
+		-DENABLE_UNINSTALL=OFF
+		-DENABLE_ZSTD=ON
+		-DUSE_BUNDLED_UTF8PROC=OFF
+	)
+
+	cmake_src_configure
+}
+
+src_compile() {
+	cmake_src_compile
+	if use test; then
+		cmake_build mongo_c_driver_tests
+		cmake_build mongo_c_driver_examples
+	fi
+}
+
+src_test() {
+	local CMAKE_SKIP_TESTS=(
+		# FIXME needs setup
+		"mongoc/fixtures/fake_kms_provider_server/start"
+		# segfaults
+		"mongoc/Topology/invalidate_server/*"
+		# needs certificates
+		"mongoc/unified/kmsProviders*"
+		# depends on AWS
+		"mongoc/client_side_encryption/unified/*"
+		# Install test
+		"mongoc/CMake/*"
+		"mongoc/pkg-config/*"
+		# Condition 'count <= 1' failed.
+		"mongoc/Client/ipv6/single"
+		# error: "Failed to connect to: localhost"
+		"mongoc/azure/imds/http/talk"
+		"mongoc/gcp/http/talk"
+	)
+	export MONGOC_TEST_OFFLINE=on
+	export MONGOC_TEST_SKIP_MOCK=on
+	if ! use test-full; then
+		export MONGOC_TEST_SKIP_LIVE=on
+	else
+		local PORT=27099
+		export MONGOC_TEST_URI="mongodb://[127.0.0.1]:${PORT}"
+		export MONGOC_ENABLE_MAJORITY_READ_CONCERN=on
+		LC_ALL=C \
+			mongod --setParameter enableTestCommands=1 \
+			--port ${PORT} --bind_ip 127.0.0.1 --nounixsocket \
+			--fork --dbpath="${T}"\
+			--logpath="${T}/mongod.log" || die
+	fi
+
+	# parallel tests cause failures.
+	cmake_src_test -j1
+	if use test-full; then
+		kill $(<"${T}/mongod.lock")
+	fi
+}
+
+src_install() {
+	if use examples; then
+		docinto examples
+		dodoc src/libmongoc/examples/*.c
+	fi
+
+	cmake_src_install
+	strip-lto-bytecode
+
+	# If USE=man, there'll be newly generated docs which we install instead.
+	if ! use man && [[ ${MONGO_C_DRIVER_DOCS_PREBUILT} == 1 ]] ; then
```

Comment:
USE="man" and USE="-man" both installing man pages is a bit weird. Id imagine it would be best to just install the prebuilt man pages unconditionally.